bloptools/bayesian/devices.py

```diff
@@ -1,218 +1,8 @@
-
-    def __repr__(self):
-        """A table showing the state of each DOF."""
-        return self.summary.__repr__()
-
-    @property
-    def summary(self) -> pd.DataFrame:
-        table = pd.DataFrame(columns=DOF_FIELDS)
-        for dof in self.dofs:
-            for attr in table.columns:
-                table.loc[dof.name, attr] = getattr(dof, attr)
-
-        # convert dtypes
-        for attr in ["readback", "lower_limit", "upper_limit"]:
-            table[attr] = table[attr].astype(float)
-
-        for attr in ["read_only", "active"]:
-            table[attr] = table[attr].astype(bool)
-
-        return table
-
-    @property
-    def names(self) -> list:
-        return [dof.name for dof in self.dofs]
-
-    @property
-    def devices(self) -> list:
-        return [dof.device for dof in self.dofs]
-
-    @property
-    def lower_limits(self) -> np.array:
-        return np.array([dof.lower_limit for dof in self.dofs])
-
-    @property
-    def upper_limits(self) -> np.array:
-        return np.array([dof.upper_limit for dof in self.dofs])
-
-    @property
-    def limits(self) -> np.array:
-        """
-        Returns a (n_dof, 2) array of bounds.
-        """
-        return np.c_[self.lower_limits, self.upper_limits]
-
-    @property
-    def readback(self) -> np.array:
-        return np.array([dof.readback for dof in self.dofs])
-
-    def add(self, dof):
-        _validate_dofs([*self.dofs, dof])
-        self.dofs.append(dof)
-
-    def _dof_read_only_mask(self, read_only=None):
-        return [dof.read_only == read_only if read_only is not None else True for dof in self.dofs]
-
-    def _dof_active_mask(self, active=None):
-        return [dof.active == active if active is not None else True for dof in self.dofs]
-
-    def _dof_tags_mask(self, tags=[]):
-        return [np.isin(dof["tags"], tags).any() if tags else True for dof in self.dofs]
-
-    def _dof_mask(self, active=None, read_only=None, tags=[]):
-        return [
-            (k and m and t)
-            for k, m, t in zip(self._dof_read_only_mask(read_only), self._dof_active_mask(active), self._dof_tags_mask(tags))
-        ]
-
-    def subset(self, active=None, read_only=None, tags=[]):
-        return DOFList([dof for dof, m in zip(self.dofs, self._dof_mask(active, read_only, tags)) if m])
-
-    def activate(self, read_only=None, active=None, tags=[]):
-        """Activate all degrees of freedom with a given tag, active status or read-only status.
-
-        For example, `dofs.activate(tag='kb')` will turn off all dofs which contain the tag 'kb'.
-        """
-        for dof in self._subset_dofs(read_only, active, tags):
-            dof.active = True
-
-    def deactivate(self, read_only=None, active=None, tags=[]):
-        """The same as .activate(), only in reverse."""
-        for dof in self._subset_dofs(read_only, active, tags):
-            dof.active = False
+from ophyd import Signal, SignalRO  # noqa F401
 
 
 class BrownianMotion(SignalRO):
```

Comment:
This class is present in [`bloptools/bayesian/dofs.py`](https://github.com/NSLS-II/bloptools/pull/52/files#diff-1eee9a2e8c39b3f341e8fa40587b83e0e2f08446388f0b48155b710fe7c23706R206) along with other classes below. Did you intend to move them all together?